v2.1/gl/conversions.go

```diff
@@ -46,6 +48,13 @@ func PtrOffset(offset int) unsafe.Pointer {
 	return unsafe.Pointer(uintptr(offset))
 }
 
+// StrFree takes a Go string and allocates a C string in the C heap using malloc.
+// It is the caller's responsibility to free it by calling returned free.
+func StrFree(str string) (cstr *uint8, free func()) {
+	ptr := unsafe.Pointer(C.CString(str))
+	return (*uint8)(ptr), func() { C.free(ptr) }
+}
+
```

Comment:
I agree that `Str` should be left as-is (as you said, it can still be used for single-string arguments).

I'm not too sure this is the right approach, though. `StrFree` here makes the assumption that you are only interested in passing one string parameter, i.e. the use-case that `Str` itself serves (we're just allocating on the heap to bypass the cgo check).

But most (all?) GL functions that will want to take a `**uint8` will be a _list of strings_ not a single string. It seems to me that instead of this, we should have a `func StrSlice(str ...string) (cstrings **uint8, free func())` instead, as otherwise we would end up needing something like this in the future too for when someone really does want to pass a string list to a GL function (as in the case of `glShaderSource`).

That said, this does clearly _workaround_ the issue at hand, and is better than the current situation (outright failure on Go 1.6), so I wouldn't protest this change if you're not up for the extra work that `StrSlice` would be to implement.